cpu/kinetis/periph/pm.c

```diff
@@ -26,17 +26,70 @@
 #define ENABLE_DEBUG (0)
 #include "debug.h"
 
-/**
- * @note    The current PM implementation is very much simplified down to only
- *          using the 'WAIT' mode. This implementation must be further expanded
- *          to make use of the available and more efficient (deep) sleep modes
- *          of the Kinetis CPUs.
- */
+/* Set to 1 to use the LEDx macros to show which sleep mode is entered */
+#define ENABLE_LEDS (0)
+
+#if ENABLE_LEDS
+#include "board.h"
+#define PM_LED(x, state) LED ## x ## _ ## state
+#else
+#define PM_LED(x, state)
+#endif
+
+/* SMC_PMCTRL_STOPM masks */
+enum {
+    SMC_PMCTRL_STOPM_STOP = 0,
+    /* 1 is reserved */
+    SMC_PMCTRL_STOPM_VLPS = 2,
+    SMC_PMCTRL_STOPM_LLS  = 3,
+    /* VLLS is not supported */
+};
+
+/** Configure which stop mode will be entered when cortexm_sleep(1) is called */
+static inline void pm_stopm(uint8_t stopm)
+{
+    SMC->PMCTRL = (SMC->PMCTRL & ~(SMC_PMCTRL_STOPM_MASK)) | SMC_PMCTRL_STOPM(stopm);
+}
+
 void pm_set(unsigned mode)
 {
+    unsigned deep = 1;
     switch (mode) {
-        case 0:
-            cortexm_sleep(0);
+        case PM_NUM_MODES:
+            return;
+        case KINETIS_PM_WAIT:
+            /* WAIT */
+            deep = 0;
+            PM_LED(0, ON);
+            break;
+        case KINETIS_PM_STOP:
+            /* STOP */
```

Comment:
Please add a description of what is still running in those states - and how you can wake up.

Reply:
I've added what I can recall for the moment. I'll have a better recollection to expand on it as I get through more of #11789. In particular I can't be very specific about the mid-range modes without some review.